src/main/java/com/auth0/utils/tokens/HS256SignatureVerifier.java

```diff
@@ -0,0 +1,22 @@
+package com.auth0.utils.tokens;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.interfaces.JWTVerifier;
+
+/**
+ * An implementation of {@code SignatureVerifier} for tokens signed with the HS256 symmetric signing algorithm.
+ */
+class HS256SignatureVerifier extends SignatureVerifier {
+
+    HS256SignatureVerifier(String secret) {
+        super(createJWTVerifier(secret), "HS256");
```

Comment:
The first parameter function would create an HS256 algorithm. The second parameter would specify the same algorithm name. I assume this is like this because of our custom exception message? But maybe there's a solution around try/catching the `AlgorithmMistmatchException`. You'll miss the expected/found values. But those could be added later to the jwt library's exception class.

Reply:
Yes it is used to generate the exception message. I see what you mean about try/catching the `AlgorithmMismatchException`. It's somewhat related to the discussion we'll have around a null verifier and whether/how to best support use cases like `mvc-commons`, so we'll discuss this as well.

Reply:
Updated to try/catch an `AlgorithmMismatchException`